game-core/src/test/java/games/strategy/engine/message/RemoteActionCodeTest.java

```diff
@@ -0,0 +1,49 @@
+package games.strategy.engine.message;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+import java.lang.reflect.Method;
+import java.util.stream.IntStream;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.aggregator.AggregateWith;
+import org.junit.jupiter.params.aggregator.ArgumentsAccessor;
+import org.junit.jupiter.params.aggregator.ArgumentsAggregationException;
+import org.junit.jupiter.params.aggregator.ArgumentsAggregator;
+import org.junit.jupiter.params.provider.CsvFileSource;
+
+public class RemoteActionCodeTest {
+
+  @ParameterizedTest
+  @CsvFileSource(resources = "/required-op-codes.txt")
+  void verifyCorrectOpCode(
+      final int opCode, @AggregateWith(MethodAggregator.class) final Method method) {
+    var remoteActionCode = method.getAnnotation(RemoteActionCode.class);
+
+    assertThat("No annotation present for " + method, remoteActionCode, is(notNullValue()));
+
+    assertThat("Invalid value for " + method, remoteActionCode.value(), is(opCode));
```

Comment:
The reason strings are not really required here, but for some reason my IntelliJ doesn't display the "correct" method name that includes the parameters toString info.